engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoFile.java

```diff
@@ -51,6 +54,25 @@ public InputStream inputStream(OpenOption[] opts) throws IOException {
     return this.truffleFile.newInputStream(opts);
   }
 
+  @Builtin.Method(name = "read_last_bytes_builtin")
+  @Builtin.WrapException(from = IOException.class, to = PolyglotError.class, propagate = true)
+  public Array readLastBytes(long n) throws IOException {
+    try (SeekableByteChannel channel = this.truffleFile.newByteChannel(Set.of(StandardOpenOption.READ))) {
+      int bytesToRead = Math.toIntExact(Math.min(channel.size(), n));
+      channel.position(channel.size() - bytesToRead);
+      ByteBuffer buffer = ByteBuffer.allocate(bytesToRead);
+      while (buffer.hasRemaining()) {
+        channel.read(buffer);
+      }
+
+      Object[] result = new Long[bytesToRead];
```

Comment:
This is certainly ineffective. All the content of the file is going to be split to 8-bytes chunks and an object (of type `java.lang.Long`) allocated for each of them. If you want this to be effective, then you can implement
```
class ArrayOverBuffer implements TruffleObject {
  final ByteBuffer buffer;
}
```
and return that object.